docs/DeveloperGuide.md

```diff
@@ -1,34 +1,422 @@
+ 5  |  CS2113  |       Tutorial       | tutorial worksheet 10  |        today 12:00PM         |  5  
++--------------------------------------------------------------------------------------------------+
+Total tasks: 5
++--------------------------------------------------------------------------------------------------+
+```
+
+![Step 1](images/Number_1.png){: .image-left .step }   
+<br>
+James will first enter the command to delete *tasks*:  
+	`delt work -a`  
+	After the input is parsed as a **delete task** command and executed, the `DeleteTaskCommand#execute()` will call `FilterCommand#createFilteredTaskList()` to create the filtered list of *tasks* containing the *description* "work". `DeleteTaskCommand#execute()` will then call its own method `DeleteTaskCommand#executeInitialDelete(filteredList)` to prepare the prompt to request James to enter the list number of the *tasks* he would like to delete.  
+
+![Step 2](images/Number_2.png){: .image-left .step }    
+<br>
+James receives the following prompt:
+
+```  
+Multiple matching tasks found.
++--------------------------------------------------------------------------------------------------+
+ NO |  MODULE  |       CATEGORY       |          TASK          |           DEADLINE           | PTY 
++--------------------------------------------------------------------------------------------------+
+ 1  |  CS1231  |      Assignment      |       group work       |      28/03/2020 06:00PM      | 20  
+ 2  |  CS2113  |      Bigger Lab      |      Big Lab Work      |        today 06:00PM         |  7  
+ 3  |  CS2113  |       Tutorial       | tutorial worksheet 10  |        today 12:00PM         |  5  
++--------------------------------------------------------------------------------------------------+
+Total tasks: 3
++--------------------------------------------------------------------------------------------------+
+
+Enter the list number(s) of the tasks to delete.
+```
+
+He proceeds to enter list numbers `2 3` as he has already completed both *tasks*.  
+After the list numbers are parsed, it will call `ListNumberPrompt#execute()`, which will prepare the prompt for the delete confirmation, and then calls `ListNumberPrompt#executePromptConfirmation()`.
+
+![Step 3](images/Number_3.png){: .image-left .step}   
+<br>
+James receives another prompt:  
+
+```  
+Confirm delete these tasks?
+tutorial worksheet 10
+Big Lab Work
+```
+
+He enters `y` to confirm the deletion.
+`DeleteConfirmationPrompt#execute()` will be called, which then calls `DeleteConfirmationPrompt#executeMultipleDelete(filteredList)` to delete James' selected *tasks* from his Task List.
+
+![Step 4](images/Number_4.png){: .image-left .step}  
+<br>
+James receives the final message:
+
+```
+SUCCESS!! Task(s) have been deleted.
+```
+
+Deletion process ends.  
+
+<br>
+
+Below is a *sequence diagram* to illustrate the above example scenario.  
+
+![Delete Command Sequence Diagram](images/Delete_Command_Sequence_Diagram.png)
```

Comment:
Here are some suggestions based on my understanding:
* 1. Method calls should have a `()`, for example use `executeInitialDelete()` instead of `executeInitialDelete`
* 2. Maybe it's better to add constuct method on the arrow representing the call to the constructor.